lib/apiservers/portlayer/swagger.json

```diff
@@ -175,8 +176,166 @@
+				      "tags": [
+					        "storage"
+				      ],
+				      "consumes": [
+					        "application/json"
+				      ],
+				      "produces": [
+					        "application/x-tar"
+				      ],
+				      "operationId": "ExportArchive",
+				      "parameters": [
+						{
+							"name": "store",
+							"in":"query",
+							"required":true,
+							"type" : "string",
+							"x-nullable": false
+						},
+						{
+							"name": "ancestorStore",
+							"in":"query",
+							"required": false,
+							"type" : "string",
+							"x-nullable": false
+						},
+						{
+							"name":"deviceID",
+							"in":"query",
+							"required": true,
+							"type" : "string",
+							"x-nullable": false
+						},
+						{
+							"name":"ancestor",
+							"in":"query",
+							"required": false,
+							"type" : "string",
+							"x-nullable": false
+						},
+						{
+							"name":"data",
+							"in":"query",
+							"required": true,
+							"type" : "boolean",
+							"x-nullable": false						
+						},
+						{
+							"name":"filterSpec",
+							"in":"query",
+							"required": false,
+							"type" : "string",
+							"x-nullable": true
+						}
+					],
+					"responses": {
+						"200": {
+							"description": "OK",
+							"schema": {
+								"format": "binary"
+							}
+						},
+					"404" : {
+						"description": "Supplied target not found",
+						"schema": {
+							"$ref": "#/definitions/Error"
+						}
+					},
+					"423": {
+						"description": "Device or resource is locked",
+						"schema": {
+							"$ref": "#/definitions/Error"
+						}
+					},
+					"500": {
+						"description": "failed to export tar archive from target"
+					},
+					"default": {
+						"description": "error",
+						"schema": {
+							"$ref": "#/definitions/Error"
+						}
+					}
+				}
+          },
+      "post": {
+          "description": "Imports a tar stream to a device at the target path",
+				  "summary": "Import tar stream to supplied target",
+				  "tags": [
+					        "storage"
+				  ],
+				  "consumes": [
+					    "application/x-tar"
+				  ],
+				  "operationId": "ImportArchive",
+				  "parameters": [
+					{
+						"name": "store",
+						"in":"query",
+						"required": true,
+						"type" : "string",
+						"x-nullable": false
+					},
+					{
+						"name":"deviceID",
+						"in":"query",
+						"required": true,
+						"type" : "string",
+						"x-nullable": false
+					},
+					{
+						"name":"filterSpec",
+						"in":"query",
+						"required": false,
+						"type" : "string",
+						"x-nullable": true
+					},
+					{
+						"name": "Archive",
+						"required": true,
+						"in": "body",
+						"schema": {
+							"type":"string",
+							"format":"binary"
+						}
+					}
+				],
+				"responses": {
+					"200": {
+						"description": "OK"
+					},
+					"404" : {
+						"description": "Supplied target not found",
+						"schema": {
+							"$ref": "#/definitions/Error"
```

Comment:
The custom tar transport we're creating in container proxy uses byte stream consumers, but this Error Model doesn't have an interface for producing byte stream. We either need to remove the message body here and produce no payloads in the storage handler, OR we need to add a byte stream producer to the error model.